### PR TITLE
Add and remove feature

### PR DESCRIPTION
**Pull request by unorjikingsley**
In this branch, I did the following functions;

1. Implement a function for adding a new task (add a new element to the array).
2. Implement a function for deleting a task (remove an element from the array).
3. Implement a function for editing task descriptions.
4. By default new tasks have the property completed set to false and the property index set to the value of the new array length.
5. Deleting a task updates all remaining items' indexes, so they represent the current list order and are unique
6. All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,28 +1,151 @@
...
-  { description: 'Read a novel', completed: true, index: 1 },
-  { description: 'Learn Webpack', completed: false, index: 2 },
-  { description: 'Do House chores', completed: false, index: 3 },
-];
+let tasks = [];
 
 function displayTasks() {
   const todoList = document.getElementById('todo-list');
   todoList.innerHTML = '';
   tasks.sort((a, b) => a.index - b.index);
```
Comment (Whoistolu) [inline src/index.js:9]: - [ ] [REQUIRED] Great job so far @unorjikingsley , however, kindly note that a part of the project requirement states that the property index should be set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). You currently have the index of the first tasks as 0, you can add 1 to the index.
### Your Design
Screenshot 2023-05-04 at 00 54 33

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @unorjikingsley 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 2]
```diff
@@ -1,28 +1,142 @@
+/* eslint-disable no-use-before-define */
```
Comment (wayungi) [inline src/index.js:1]: - [ OPTIONAL] My kind suggestion is that you should always avoid disabling linters. Linters ensure that we write good & uniform quality code. 😄

File: src/index.js [hunk 3]
```diff
@@ -1,28 +1,142 @@
...
+      task.index = i;
+    });
+    saveTasks();
+    displayTasks();
+  });
+});
+
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
```
Comment (wayungi) [inline src/index.js:107]: - [ OPTIONAL] I think you should write arrow functions for all your function. They are simpler, clear, and easy to read. 
__An arrow function expression is a compact alternative to a traditional function__

File: src/index.js (hunk 1 above)
Comment (wayungi) [reply src/index.js:9]: - [ ] Great job fixing the index numbering ✔️

**Review by wayungi — APPROVED** [review]
Hi @unorjikingsley , 

image

Great work, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**Remember: Every finish line 🏁  is the beginning of a new race 🐎** 

## Highlights
- [ ] All linter checks are passing ✔️ 
- [ ]  Correct GitHub flow ✔️ 

## Optional suggestions
- [ OPTIONAL] A good PR description includes only a brief summary of features you have added to the project. You do not need to include project requirements because the reviewers already know them.
inst

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**